crates/matrix-sdk/src/authentication/oauth/mod.rs

```diff
@@ -27,53 +27,63 @@
 //!
-//! Registration is only required the first time a client encounters a
-//! homeserver.
+//! Clients must register with the homeserver before
 //!
 //! Note that only public clients are supported by this API, i.e. clients
 //! without credentials.
 //!
-//! If the server supports dynamic registration, it can be done by using
-//! [`OAuth::register_client()`]. After registration, the client ID should be
-//! persisted and reused for every session that interacts with that same server.
+//! The registration step can be done automatically by providing a
+//! [`ClientRegistrationData`] to the login method.
 //!
-//! If dynamic registration is not available, the homeserver should document how
-//! to obtain a client ID.
-//!
-//! To provide the client ID and metadata if dynamic registration is not
-//! available, or if the client is already registered with the issuer, call
-//! [`OAuth::restore_registered_client()`].
+//! If the server supports dynamic registration, registration can be performed
+//! manually by using [`OAuth::register_client()`]. If dynamic registration is
+//! not available, the homeserver should document how to obtain a client ID. The
+//! client ID can then be provided with [`OAuth::restore_registered_client()`].
 //!
 //! # Login
 //!
-//! Before logging in, make sure to register the client or to restore its
-//! registration, as it is the first step to know how to interact with the
-//! issuer.
+//! Currently, two login methods are supported by this API.
+//!
+//! ## Login with the Authorization Code flow
 //!
-//! With OAuth 2.0, logging into a Matrix account is simply logging in with a
-//! predefined scope, part of it declaring the device ID of the session.
+//! This method requires to open a URL in the end-user's browser where they will
+//! be able to log into their account in the server's web UI and grant access to
+//! their Matrix account.
 //!
 //! [`OAuth::login()`] constructs an [`OAuthAuthCodeUrlBuilder`] that can be
 //! configured, and then calling [`OAuthAuthCodeUrlBuilder::build()`] will
-//! provide the URL to present to the user in a web browser. After
-//! authenticating with the server, the user will be redirected to the provided
-//! redirect URI, with a code in the query that will allow to finish the
-//! login process by calling [`OAuth::finish_login()`].
+//! provide the URL to present to the user in a web browser.
 //!
-//! # Persisting/restoring a session
+//! After authenticating with the server, the user will be redirected to the
+//! provided redirect URI, with a code in the query that will allow to finish
+//! the login process by calling [`OAuth::finish_login()`].
+//!
+//! If the login needs to be cancelled before its completion,
+//! [`OAuth::abort_login()`] should be called to clean up the local data.
+//!
+//! ## Login by scanning a QR Code
+//!
+//! This method requires to have another logged-in Matrix device that can
+//! display a QR Code. This is a method that uses the Device Code flow
```

Comment:
Would be nice to link to what the `Device Code flow` is as well.

Reply:
I added links to the MSC and RFC.